python/feldera/enums.py

```diff
@@ -1,33 +1,177 @@
 from enum import Enum
 
 
-class CompilationProfile(Enum):
+# https://stackoverflow.com/questions/50473951/how-can-i-attach-documentation-to-members-of-a-python-enum/50473952#50473952
+class DocEnum(Enum):
+    """
+    :meta private:
+    """
+
+    def __new__(cls, value, doc=None):
+        self = object.__new__(cls)  # calling super().__new__(value) here would fail
+        self._value_ = value
+        if doc is not None:
+            self.__doc__ = doc
+        return self
+
+
+class CompilationProfile(DocEnum):
     """
     The compilation profile to use when compiling the program.
     """
 
-    SERVER_DEFAULT = None
+    SERVER_DEFAULT = None, "The compiler server default compilation profile."
+
+    DEV = "dev", "The development compilation profile."
+
+    UNOPTIMIZED = "unoptimized", "The unoptimized compilation profile."
+
+    OPTIMIZED = "optimized", "The optimized compilation profile, the default for this API."
+
+
+class BuildMode(DocEnum):
+    CREATE = 1
+    GET = 2
+    GET_OR_CREATE = 3
+
+
+class PipelineStatus(DocEnum):
     """
-    The compiler server default compilation profile.
+    Represents the state that this pipeline is currently in.
+
+    .. code-block:: text
+
+        Shutdown     ◄────┐
+        │         │
+        /deploy   │       │
+        │   ⌛ShuttingDown
+        ▼         ▲
+        ⌛Provisioning    │
+        │         │
+        Provisioned        │
+        ▼         │/shutdown
+        ⌛Initializing     │
+        │        │
+        ┌────────┴─────────┴─┐
+        │        ▼           │
+        │      Paused        │
+        │      │    ▲        │
+        │/start│    │/pause  │
+        │      ▼    │        │
+        │     Running        │
+        └──────────┬─────────┘
+                   │
+                   ▼
+                Failed
+   """
+
+    UNINITIALIZED = 1, """
```

Comment:
`NON_EXISTENT` / `DOES_NOT_EXIST` / `NOT_CREATED` / `NOT_FOUND` or so is more appropriate naming, as both `SHUTDOWN `and `PROVISIONING` are states that are "uninitialized".